mlir/lib/Dialect/Rock/Transforms/Regularize.cpp

```diff
@@ -196,26 +202,33 @@ struct PushTransformsUpRewritePattern
     // find writer and readChains (must be a transform)
     bool hasTransforms = false;
     Operation *writer = nullptr;
+    Operation *fusor = nullptr;
     SmallVector<SmallVector<Operation *>> readChains;
     for (auto &use : buffer.getUses()) {
       Operation *useOp = use.getOwner();
       SmallVector<Operation *> chain;
       bool isWriter = collectChain(buffer, useOp, chain);
+
+      bool isFusor = isFusorOp(chain.back());
       if (isWriter) {
         assert(writer == nullptr);
         writer = useOp;
+      }
+      if (isFusor) {
```

Comment:
I'm seeing a correctness issue, it goes like this

```
%alloc1 = alloc() : T
rock.gemm %alloc1 = ...
%alloc2 = alloc() : T
linalg.generic ins (%alloc1, ...) outs(%alloc2) { ... }
%v = view(%alloc2) : T to U
memref.copy %v, %argN : U to U
```

The regularization invariant for output fusions says this must be
```
%alloc1 = alloc() : T
gemm %alloc1 = ...
%alloc2_new = alloc() : U
%alloc2_like_old = [inverse view](%alloc2_new) : U to T
la.generic ins(%alloc1, ...) outs(%alloc2_like_old) { ... }
memref.copy %alloc2_new, %argN : U to U
```

Your definition of `fusor` won't cause the rewrite to fire like it used to.